common/models/role-mapping.js

```diff
@@ -24,9 +24,11 @@ module.exports = function(RoleMapping) {
    * @param {Application} application
    */
   RoleMapping.prototype.application = function(callback) {
+    var registry = this.constructor.registry || loopback.registry;
+
     if (this.principalType === RoleMapping.APPLICATION) {
       var applicationModel = this.constructor.Application ||
-        loopback.getModelByType(loopback.Application);
+        registry.getModelByType(loopback.Application);
```

Comment:
As I explained elsewhere, you must not use `loopback.Application` here, as it is potentially defined in a different registry.